### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ValeriyPetryna**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+import { HeaderComponent } from './shared/components/header/header.component';
+import { LikeComponent } from './shared/components/like/like.component';
+
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    MainLayoutComponent,
+    PostListComponent,
```
Comment (MMaksymivUa) [inline src/app/app.module.ts:21]: Create separate module for Posts, and maybe(optional) shared/core module for layout

File: src/app/main-layout/main-layout.component.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    display: flex;
+    justify-content: space-between;
+    margin: 5px;
+    &__blog {
+        margin-right: 10px;
+        flex-grow: 8;
+        border: 1px solid gray;
+    }
+    &__aside {
+        background-color: rgb(207, 196, 196);
```
Comment (MMaksymivUa) [inline src/app/main-layout/main-layout.component.scss:11]: convert to hex and define a variable

File: src/app/posts/components/post-form/post-form.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <form class="modal-body" [formGroup]="postForm" (ngSubmit)="onSubmit()">
+        <div class="postForm__item">
+            <input formControlName="author" class="postForm__item-input" type="text" placeholder="Article author"/>
+            <p class="validation" *ngIf="showValidationError('author')">Field author is required!</p>
+            <p class="validation" *ngIf="showValidationError('author', 'minlength')">Author field must be at least 4 characters long.</p>
+        </div>
+        <div class="postForm__item">
+            <input formControlName="title" class="postForm__item-input" type="text" placeholder="Article title"/>
+            <p class="validation" *ngIf="showValidationError('title')">Field title is required!</p>
+            <p class="validation" *ngIf="showValidationError('title', 'minlength')">Title field must be at least 5 characters long.</p>
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-form/post-form.component.html:11]: bad approach, add check error function to the class. You can store a Map<Error, ErrorDescription> and show the error label by using map[error]

File: src/app/posts/components/post-form/post-form.component.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        </div>
+        <div class="postForm__item">
+            <input formControlName="content" class="postForm__item-input" type="text" placeholder="Article content"/>
+            <p class="validation" *ngIf="showValidationError('content')">Field content is required!</p>
+            <p class="validation" *ngIf="showValidationError('content', 'minlength')">Content field must be at least 10 characters long.</p>
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-form/post-form.component.html:16]: same here

File: src/app/posts/components/post-form/post-form.component.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+        input[type=text] {
+            width: 100%;
+            padding: 12px 20px;
+            margin: 8px 0;
+            box-sizing: border-box;
+            border: 2px solid rgb(197, 177, 177);
+            border-radius: 4px;
+            background-color: #bcffe6;
+            color: rgb(34, 33, 33);
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-form/post-form.component.scss:17]: hex

File: src/app/posts/components/post-form/post-form.component.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        button {
+            background-color: #4CAF50;
+            border: none;
+            color: white;
+            padding: 15px 32px;
+            text-align: center;
+            text-decoration: none;
+            display: inline-block;
+            font-size: 16px;
+        }
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-form/post-form.component.scss:29]: define general buttons for the project style.scss or in mixin file

File: src/app/posts/components/post-form/post-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  constructor(private formBuilder: FormBuilder, private postService: PostFormService, private modal: NgbModal) {
+      this.postForm = this.formBuilder.group({
+        author: ['', [Validators.required, Validators.minLength(4), Validators.maxLength(50)]],
+        title: ['', [Validators.required, Validators.minLength(5)]],
+        content: ['', [Validators.required, Validators.minLength(10)]]
+      })
+   }
+
+  onSubmit() {
+    if(this.postForm.status === 'VALID') {
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-form/post-form.component.ts:24]: there are should be a  special filed valid in postForm

File: src/app/posts/components/post-list/post-list.component.html [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<app-post
+    class="post"
+    *ngFor = "let i of posts"
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-list/post-list.component.html:3]: (optional) use better naming: let post of posts

File: src/app/posts/components/post-list/post-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      title: 'Accessibility of Remote Meetings',
+      content: 'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Corporis quibusdam minus sapiente facere ab quae libero necessitatibus quod? Voluptas, neque.',
+      date: new Date(),
+    }
+  ];
+
+  constructor(private postService: PostFormService) { }
+
+  ngOnInit(): void {
+    this.postServiceSubscribe();
```
Comment (MMaksymivUa) [inline src/app/posts/components/post-list/post-list.component.ts:32]: no need of function, simply use
```this.postServiceSubscription = this.postService.getPostData().subscribe((data: Post) => this.posts.push(data));```

File: src/app/posts/components/post/post.component.html [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<article class="post">
```
Comment (MMaksymivUa) [inline src/app/posts/components/post/post.component.html:1]: *ngIf="post"
check if post exist or use for post reference "?" mark like
```
{{post?.author}}
```

File: src/app/posts/components/post/post.component.html [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+            <div class="post__author-info">
+                <div class="post__author-name">{{ post.author }}</div>
+                <div class="post__date">{{ post.date | date : 'longDate' }}</div>
+            </div>
+        </div>     
+        <div class="post__like">
+            <img src="./assets/images/like.png" alt="likes" width="50" height="50">
+        </div>
+    </div>
+    <div class="post__title"><strong>{{ post.title }}</strong></div>
```
Comment (MMaksymivUa) [inline src/app/posts/components/post/post.component.html:16]: <b></b>
strong is outdated standard

File: src/app/posts/components/post/post.component.scss [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        margin: 5px;
+    }
+
+    &__author {
+        display: flex;
+        letter-spacing: 1px;
+        font-family: cursive;
```
Comment (MMaksymivUa) [inline src/app/posts/components/post/post.component.scss:13]: ???

File: src/app/posts/services/post-form.service.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+
+export class PostFormService {
+  private postDataStream = new Subject<Post>();
+  constructor() { }
+
+  public updatePostData(data: Post): void {
+    this.postDataStream.next(data);
```
Comment (MMaksymivUa) [inline src/app/posts/services/post-form.service.ts:14]: how about of 
```
this.postDataStream.next([...this.postDataStream.getCurrentValue(), data]);
```
But convert it to BehaviourSubject

File: src/app/shared/components/header/header.component.scss [hunk 14]
```diff
@@ -0,0 +1,21 @@
+.header {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    margin: 20px 0 5px 20px;
+
+    &__title {
+        color: rgb(33, 226, 33);
```
Comment (MMaksymivUa) [inline src/app/shared/components/header/header.component.scss:9]: hex

Comment (ValeriyPetryna) [reply src/app/shared/components/header/header.component.scss:9]: Fixed in new PR Feature/hw3

File: src/app/posts/services/post-form.service.ts (hunk 13 above)
Comment (ValeriyPetryna) [reply src/app/posts/services/post-form.service.ts:14]: Fixed in new PR Feature/hw3, thanks for the advice!

File: src/app/posts/components/post/post.component.html (hunk 11 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post/post.component.html:16]: Removed in new PR Feature/hw3

File: src/app/posts/components/post/post.component.html (hunk 10 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post/post.component.html:1]: Fixed in new PR Feature/hw3, added ngIf in the article element

File: src/app/posts/components/post-list/post-list.component.ts (hunk 9 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post-list/post-list.component.ts:32]: Fixed in new PR Feature/hw3

File: src/app/posts/components/post-list/post-list.component.html (hunk 8 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post-list/post-list.component.html:3]: Fixed in new PR Feature/hw3

File: src/app/posts/components/post-form/post-form.component.ts (hunk 7 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post-form/post-form.component.ts:24]: Fixed in new PR Feature/hw3, thanks

File: src/app/posts/components/post-form/post-form.component.scss (hunk 6 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post-form/post-form.component.scss:29]: Replaced with default material buttons for now

File: src/app/posts/components/post-form/post-form.component.scss (hunk 5 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post-form/post-form.component.scss:17]: Done, Fixed in new PR Feature/hw3

File: src/app/main-layout/main-layout.component.scss (hunk 2 above)
Comment (ValeriyPetryna) [reply src/app/main-layout/main-layout.component.scss:11]: Done, Fixed in new PR Feature/hw3

File: src/app/app.module.ts (hunk 1 above)
Comment (ValeriyPetryna) [reply src/app/app.module.ts:21]: Fixed in new PR Feature/hw3, added blog, shared header, and post form modules, removed some extra layers, renamed some components

File: src/app/posts/components/post/post.component.scss (hunk 12 above)
Comment (ValeriyPetryna) [reply src/app/posts/components/post/post.component.scss:13]: My bad, forgot to move these text-format props to the post__author-name class, moved into the correct one in the new PR Feature/hw3.

**Pull request merged**